GNUmakefile

```diff
@@ -61,6 +61,7 @@ GO_BUILD_TAG?=consul-build-go
 UI_BUILD_TAG?=consul-build-ui
 BUILD_CONTAINER_NAME?=consul-builder
 CONSUL_IMAGE_VERSION?=latest
+ENVOY_VERSION?=1.24.6
```

Comment:
This is the version we use in compat test https://github.com/hashicorp/consul/blob/30eee13cb97cd78bf3f74b747c17a1ea814c544a/test/integration/consul-container/libs/service/common.go#L18

bump the version to 1.24.6